src/features/auth/context.js

```diff
@@ -19,10 +21,12 @@ const initialState = () => {
 };
 
 // eslint-disable-next-line react/prop-types
-export const AuthProvider = ({ httpClient, children }) => {
-  const [state, setState] = useState(initialState);
+export const AuthProvider = ({ children, prepopulatedState, httpClient }) => {
+  const [state, setState] = useState(prepopulatedState || initialState);
   const { user, session, isLoading } = state;
 
+  const { guestLocale } = useGuestLocale();
```

Comment:
I think it would be better to receive the `guestLocale` as a prop for when we move this to a separate library